Enforce code style

matrei opened the pull request:
This PR adds CodeNarc and Checkstyle Gradle Plugins to enforce code styles.
It adds a Gradle task `codeStyle` to run all `codenarc*` and `checkstyle*` tasks exclusively.

It also adds a Gradle property check for `skipCodeStyle` to skip code style checking when running `build` or `check`.
Existing GitHub workflows are updated to use this property and a new workflow is added that does only code style checks.

```console
# Run all style checks locally
./gradlew codeStyle

# Include style checks during build/check
./gradlew check
./gradlew build

# Skip style checks (e.g., for quick iterations or CI main pipelines)
./gradlew check -PskipCodeStyle
./gradlew build -PskipCodeStyle
```

matrei committed:
style: conform to style rules

matrei committed:
fix: add style enforcement

matrei committed:
ci: run style checks in dedicated workflow

Style/formatting checks are executed only by the new workflow,
keeping the main pipelines focused on build/test.

matrei committed:
fix: `skipCodeStyle` task condition

matrei committed:
ci: add report upload

matrei committed:
ci: set correct report directory

matrei committed:
chore: trigger code style errors

To test if reports are uploaded to the workflow
summary page.

matrei committed:
ci: specify code style reports more granular

matrei committed:
ci: add `--continue` to get all codestyle errors

matrei committed:
ci: set correct `skipCodeStyle` condition for `grails-gradle`

matrei committed:
ci: stop using Gradle --continue for code style checks

Code style tools (Checkstyle/CodeNarc) generate reports even when clean.
Fail fast on style violations instead of continuing and producing
pages of reports. Developers can still use `--continue` locally when needed.

matrei committed:
Revert "chore: trigger code style errors"

This reverts commit 12a1270e94316ff5ef6623cf945c0a0d9df673ec.

jamesfredley commented on .github/workflows/codestyle.yml:
```diff
@@ -0,0 +1,82 @@
+          path: |
+            **/build/reports/checkstyle/
+            **/build/reports/codenarc/
+  check_gradle_plugin_projects:
```
Do we need a job here for grails-forge?

matrei replied:
Good catch!

matrei committed:
style: conform to code style

matrei committed:
refactor(build): dry up `code-style-config`

matrei committed:
fix(build): align code style checking in `grails-forge`

Grails Forge is still using its own Checkstyle rules
after this change.

matrei committed:
ci: add `grails-forge` code style checking

matrei committed:
ci: fix typos

jdaugherty commented on .github/workflows/gradle.yml:
```diff
@@ -256,7 +275,9 @@ jobs:
-          -Dorg.gradle.internal.publish.checksums.insecure=true --no-build-cache --rerun-tasks
+          --no-build-cache
+          --rerun-tasks
+          -Dorg.gradle.internal.publish.checksums.insecure=true
```
the insecure publishing is no longer required since we use the grails publish plugin and it sets this up by default

jdaugherty reviewed (approved):
Can we update DEVELOPMENT.md for the new properties & codestyle workflows?

matrei committed:
ci: remove redundant gradle param

`-Dorg.gradle.internal.publish.checksums.insecure=true`
is no longer required as it is set by the publish
plugin by default.

matrei committed:
docs: add `skipCodeStyle` to `DEVELOPMENT.md`

matrei committed:
docs: update `DEVELOPMENT.md`

matrei committed:
docs: update `DEVELOPMENT.md`

Add info about useful custom Gradle tasks.

matrei committed:
docs: update `CONTRIBUTING.md`

Update the section on code style.

matrei committed:
Merge branch '7.0.x' into enforce-code-style
